spec/expression.dd

```diff
@@ -1100,6 +1100,60 @@ $(GNAME CastExpression):
         (foo) - p;      // subtract p from foo
         -------------
 
+$(H4 $(LNAME2 cast_basic_data_types, Basic Data Types))
+    $(P There are some cases where the $(DDSUBLINK spec/type, implicit-conversions, implicit conversion)
+        cannot be applied, so cast may be used to force the type system to accept the reinterpretation.
+    )
+
+    $(P This situation may occur, for example, when trying to assign a value stored on four bytes
+        to a variable stored on one byte. Without a cast, the following error will be displayed:
+        "cannot implicitly convert expression a of type int to byte".
+    )
+
+    -------------
+    int a;
+    byte b = a; // error
+    -------------
+
+    $(P When the destination type is smaller than the source type,
+        the value is truncated to the destination size.
+    )
+
+        $(SPEC_RUNNABLE_EXAMPLE_RUN
+            -------------
+            int a = 64389; // 00000000 00000000 11111011 10000101
+            byte b = cast(byte) a;       // 10000101
+            ubyte c = cast(ubyte) a;     // 10000101
+            short d = cast(short) a;     // 11111011 10000101
+            ushort e = cast(ushort) a;   // 11111011 10000101
+
+            writeln(b);
+            writeln(c);
+            writeln(d);
+            writeln(e);
+            -------------
+        )
+
+    $(P Casting from a narrower type to a wider type could be done automatically,
```

Comment:
What does "could be done automatically mean" ? Is it done or is it not done?

I assume you want to say: "For integral types casting from a narrower type to a wider type is done by performing sign extension."